File: cronplusservice/Models/TaskConfig.cs
#if WINDOWS
using System.Drawing;
using System.Drawing.Printing;
#endif
using System;
using System.IO;
using SurrealDb.Net.Models;
using System.Text.Json.Serialization;

namespace CronPlus.Models;

public enum TriggerType
{
    FileCreated,
    FileRenamed,
    Time,
    Interval
}

public class TaskConfig : Record
{
    // Default constructor required for JSON deserialization
    public TaskConfig()
    {
        // Initialize default values
        triggerType = TriggerType.FileCreated;
        sourceFolder = string.Empty;
        taskType = string.Empty;
        destinationFolder = string.Empty;
        createdAt = DateTime.UtcNow;
        updatedAt = DateTime.UtcNow;
    }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public TriggerType triggerType { get; set; }
    
    public string? sourceFolder { get; set; }
    
    public string? destinationFolder { get; set; }
    
    public string? taskType { get; set; }
    
    public string? sourceFile { get; set; }
    
    public string? destinationFile { get; set; }
    
    public string? time { get; set; }
    
    public int interval { get; set; }
    
    public string? printerName { get; set; }
    
    public string? archiveDirectory { get; set; }

    public DateTime? createdAt { get; set; }

    public DateTime? updatedAt { get; set; }
}

File: cronplusui/src/types/taskConfig.ts
export enum TriggerType {
  FileCreated = 'FileCreated',
  FileRenamed = 'FileRenamed',
  Time = 'Time',
  Interval = 'Interval'
}

export interface TaskConfig {
  id?: string;
  triggerType: TriggerType;
  sourceFolder: string;
  destinationFolder: string;
  taskType: string;
  sourceFile?: string;
  destinationFile?: string;
  time?: string;
  interval?: number;
  printerName?: string;
  archiveDirectory?: string;
  createdAt?: Date;
  updatedAt?: Date;
}
